### ISSUE TITLE
Read-only Royale Interface for Domino Visual Editor

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Moonshine-IDE#1039]
This is a sub-task of #675.  @JustinProminic identified this as a priority, so I pulled it out into a new issue.

One of the early goals of #675 is to generate a read-only interface for some Prominic database.  This interface will then be used in MyAccount, with some customization.  The PerformanceStats database has useful statistics for customers, but we it is not something that customers need to submit updates to.  We need to have a clean read-only interface for this.

Currently, the generated Royale interface looks like this in read-only mode.  The fields are disabled, but they are still fields:

![image](https://user-images.githubusercontent.com/24532731/171503041-11337697-5a8c-41ac-bcf6-22ed76b4f08f.png)

Instead, we want this to look more like the Notes client, where the field values are displayed as labels:

![image](https://user-images.githubusercontent.com/24532731/171503065-ca5b8d6a-a28a-4bcc-b9e9-838006c8dd7d.png)

We do something similar already for the My Details interface:

![image](https://user-images.githubusercontent.com/24532731/171503087-6f674790-40ed-48e2-bc77-3ee646b1df94.png)

Some field types that need special log:
- Richtext and RichTextLite should be displayed as HTML.  
- "keyword" type fields (like the checkbox in the example above) should generally be displayed as fields/controls still.
- Multi-value fields should be displayed with separators.  The separator is configured in `Multi-Value Options > Display separate values with`

There are probably other special cases I am forgetting.  Let me know when you find this, and I'll add some clarification.

**Comment by piotrzarzycki21** [comment]
I did pushed changes which after activating read only mode in form disable text inputs and calendar component in a way as it is required in this issue. 
I still need to figure out how to make this happen with JoditEditor - this part is still under development. 

In order to test it:
1) Export Domino project to Royale
2) Change property **isDisabled = true** in any form
3) Build project using Royale

**Comment by piotrzarzycki21** [comment]
I have found option called "inline" in JoditEditor it shrinks nicely editor when it's in read mode. Everything look a lot better now. We need to decide if it's acceptable.

**Comment by JoelProminic** [comment]
I did a quick test with read-only mode in the latest Royale interface.  I see that the fields are disabled in the Edit form but the Add button is still visible and enabled for multivalue fields:
![image](https://user-images.githubusercontent.com/24532731/194430392-d4c0ec0e-93a7-41dd-8179-6a25967d2a18.png)

The Add button opens the form, but the fields are disabled.

The Delete button is enabled and can still trigger the action.

**Comment by piotrzarzycki21** [comment]
I have take care of disabling buttons in read mode Save, buttons in MultiValueList as well. However what about buttons on list not form - should I disable such buttons like Add, Remove ?

**Comment by JoelProminic** [comment]
Yeah, Add and Remove should be disabled.  Hiding them would probably be better.

**Comment by piotrzarzycki21** [comment]
@JoelProminic This issue require more work due to more requirements from Justin. Can you describe what I should improve ?

**Comment by piotrzarzycki21** [comment]
> @JoelProminic This issue require more work due to more requirements from Justin. Can you describe what I should improve ?

I see issue #1110 which probably is the way to go.

**Comment by piotrzarzycki21** [comment]
I'm closing this general issue in favour of #1110

**Comment by JoelProminic** [comment]
This is actually a different use case from #1110.  See [this comment](https://github.com/Moonshine-IDE/Moonshine-IDE/issues/1110#issuecomment-1293783849) for more information.  

This issue is focused on the case where we want to provide data that should never be edited by the end user.   So, the Add, Edit, and Remove buttons should not be visible.  (The Create, Read, and Delete agents should also be removed, but that is outside the scope of this issue.  For now, the user would delete the agents manually in the Java project).

We found some visual bugs with at least the rich text fields in yesterday's testing.  I'll add some screenshots when I have a chance.

**Comment by JoelProminic** [comment]
Some problems noticed in yesterday's meeting:
- Text fields are truncated
- Number fields are aligned incorrectly
- Overlapping text in multi-line rich text fields.

![image](https://user-images.githubusercontent.com/24532731/198350795-06ff0d39-228c-450d-8974-4c0ec06061ac.png)

![image](https://user-images.githubusercontent.com/24532731/198350774-cbcca33e-5e13-4a5a-aa22-8edbefe6231d.png)

**Comment by piotrzarzycki21** [comment]
> * Number fields are aligned incorrectly

I believe that part is an issue relate to @feather812002 issue -> https://github.com/Moonshine-IDE/NSFConverterPortalProject/issues/20

**Comment by piotrzarzycki21** [comment]
> * Overlapping text in multi-line rich text fields.

I have fixed this one. It should be ready to test with next Moonshine build.

**Comment by piotrzarzycki21** [comment]
Rich text is still in read only mode when we enter to edit mode.

**Comment by piotrzarzycki21** [comment]
> * Text fields are truncated

@JoelProminic I have pushed changes related to that issue. However it won't fix it. It turns out that we didn't take into account for Royale export width/height of component based on what is in intermediate XML. 

When intermediate XML of Paragraph has some width/height setup we have been omitting that. Now it's being taken into account.  

Now most of the code after translation looks like that:
```
<j:HGroup> <!-- This is Paragraph - no sizes setup because nothing was setup in intermediate xml -->
              <j:TextInput text="{itemVO.Name}" className="readOnlyInput" change="{itemVO.Name = event.currentTarget.text;}" width="100" height="30"> <!-- width/height come from intermediate xml -->
                <j:beads>
                  <j:Disabled disabled="{isDisabled}"/>
                </j:beads>
              </j:TextInput>
            </j:HGroup>
```
Values of width/height coming from intermediate xml - I believe if we wanted to have more clean interface we should decide whet components have at least setup width="100%" - if user didn't setup explicit width. 

Thoughts ?

**Comment by JoelProminic** [comment]
The Field and Paragraph width are being discussed in other issues;
https://github.com/Moonshine-IDE/NSFConverterPortalProject/issues/22
https://github.com/Moonshine-IDE/NSFConverterPortalProject/issues/23
